src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
+"""Pylint checker ensure correct usage of datetime module, especially around aware/naive objects."""
 
-"""checker to ensure correct datetime usage with aware objects in datetime module."""
+from typing import Any
+
+import astroid  # type: ignore
+import pylint  # type: ignore
+from pylint.checkers import BaseChecker  # type: ignore
+
+# function calls to fromisoformat, strptime and fromisoformat can't be checked - they parse strings
+# if the specified timezone is None, still allowed although they produce naive objects
+
+
+class DatetimeChecker(BaseChecker):  # type: ignore
+    """class for custom pylint checker."""
+
+    name = "pylint-datetime"
+    priority = -1
+    msgs = {
+        "W9999": (
+            "timedelta() called without keyword arguments",
+            "timedelta-no-keyword-args",
+            "timedelta() called without keyword arguments.",
+        ),
+        "W9998": (
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+            "timezone-no-argument",
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+        ),
+        "W9997": (
+            'Function call to "%s" can only produce naive objects',
+            "naive-datetime-call",
+            'Function call to "%s" can only produce naive objects, aware objects necessary',
+        ),
+        "W9996": (
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+            "naive-datetime-no-timezone",
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+        ),
+    }
+
+    def __init__(self, linter: pylint.lint.PyLinter = None) -> None:
+        """Initialize checker.
+
+        Args:
+            linter: instance of pylint linter.
+        """
+        super().__init__(linter)
+
+    def timedelta_with_keywords(
+        self, node: astroid.node_classes.NodeNG, name: str
+    ) -> None:
+        """Check calls to timedelta() and datetime.timedelta() for keyword arguments.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name == "timedelta":
+            for _ in node.args:
+                self.add_message("timedelta-no-keyword-args", node=node)
+            if not node.args and not node.keywords:
+                self.add_message("timedelta-no-keyword-args", node=node)
+
+    # check for positional arguments cheesy, uses fact, that they have one Attribute Argument.
+
+    def timezone_argument(self, node: astroid.node_classes.NodeNG, name: str) -> None:
+        """Check calls to datetime, now, fromtimestamp, astimezone and time for timezone arg.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name in ("datetime", "now", "fromtimestamp", "astimezone"):
```

Comment:
I would move this to the method call site. So

```py
if func_name in ("datetime", "now", "fromtimestamp", "astimezone"):
    self.check_function_called_with_timezone(node, func_name)
```

This also applies to the rest.